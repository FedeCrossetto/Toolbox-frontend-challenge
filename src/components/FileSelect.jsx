import React from 'react';
import Form from 'react-bootstrap/Form';

const FileSelect = ({ list, selectedFile, setSelectedFile }) => {
  return (
    <Form.Select
      style={{ width: '90%', margin: 'auto' }}
      value={selectedFile}
      onChange={(e) => setSelectedFile(e.target.value)}
    >
      <option value="" hidden>
        Select a file
      </option>
      {list.map((file, index) => (
        <option key={index}>{file}</option>
      ))}
    </Form.Select>
  );
};

export default FileSelect;