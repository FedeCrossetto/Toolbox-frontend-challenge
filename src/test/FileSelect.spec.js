import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import FileSelect from '../components/FileSelect';

describe('FileSelect component', () => {
  const list = ['file1', 'file2', 'file3'];
  const setSelectedFile = jest.fn();

  test('renders correctly', () => {
    const { container } = render(
      <FileSelect list={list} selectedFile={''} setSelectedFile={setSelectedFile} />
    );
    expect(container).toMatchSnapshot();
  });

  test('selects file', () => {
    const { getByRole } = render(
      <FileSelect list={list} selectedFile={''} setSelectedFile={setSelectedFile} />
    );
    const select = getByRole('combobox');
    expect(select).toBeInTheDocument();
    fireEvent.change(select, { target: { value: 'file1' } });
    expect(setSelectedFile).toHaveBeenCalledWith('file1');
  });
});