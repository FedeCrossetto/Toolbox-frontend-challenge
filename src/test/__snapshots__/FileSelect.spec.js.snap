// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileSelect component renders correctly 1`] = `
<div>
  <select
    class="form-select"
    style="width: 90%; margin: auto;"
  >
    <option
      hidden=""
      value=""
    >
      Select a file
    </option>
    <option>
      file1
    </option>
    <option>
      file2
    </option>
    <option>
      file3
    </option>
  </select>
</div>
`;
